crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -2008,11 +2008,10 @@ unsafe impl<T: Component> ReadOnlyQueryData for Has<T> {}
 pub struct AnyOf<T>(PhantomData<T>);
 
 macro_rules! impl_tuple_query_data {
```

Comment:
I think we misunderstood each other here :)

`impl QueryData for (...)` should be annotated with `fake_variadic`
`impl WorldQuery for AnyOf<(...)>` should not be annotated, so the macro no longer needs to acccept attributes via `#[$meta:meta])*`.

<sup>You can probably revert the entire file using `git restore --source=upstream/main crates/bevy_ecs/src/query/fetch.rs` assuming you have run `git fetch upstream` beforehand and the remote pointing to the bevy repo is called `upstream`.</sup>